datasquare_test.go

```diff
@@ -423,7 +423,10 @@ func computeRowProof(ds *dataSquare, x uint, y uint) ([]byte, [][]byte, uint, ui
 	data := ds.row(x)
 
 	for i := uint(0); i < ds.width; i++ {
-		tree.Push(data[i])
+		err := tree.Push(data[i])
+		if err != nil {
+			return nil, nil, 0, 0, err
```

Comment:
[Suggestion] I'd suggest adding a godoc for this function `computeRowProof`.  It makes it easier to interpret the outputs.

Reply:
+1 I agree. TBH I like named return parameters because they help document the function. Ex:

```go
func computeRowProof(ds *dataSquare, x uint, y uint) (merkleRoot []byte, proof [][]byte, proofIndex uint, numLeaves uint, err error) {
```

but our team has mixed opinions on it so I'll create a follow up issue to add godoc for this.

Reply:
Created https://github.com/celestiaorg/rsmt2d/issues/212